src/lib.rs

```diff
@@ -44,162 +44,51 @@ pub mod lightningrpc;
 pub mod requests;
 pub mod responses;
 
-use strason::Json;
 // Re-export error type
 pub use error::Error;
 // Re-export high-level connection type
 pub use lightningrpc::LightningRPC;
 
-#[derive(Debug, Clone, PartialEq, Deserialize, Serialize)]
+use serde::Serialize;
+
+#[derive(Debug, Clone, PartialEq, Serialize)]
 /// A JSONRPC request object
-pub struct Request {
+pub struct Request<'f, T: Serialize> {
```

Comment:
I piggy backed these structs from the `jsonrpc` crate, but for now that's not a good idea at least until [it's client-agnostic](https://github.com/apoelstra/rust-jsonrpc/pull/20).

Reply:
So you mean as soon as it's client agnostic we could build a backend that doesn't speak HTTP but just connects to a unix socket and then use it here instead of the home grown solution?